src/main/java/sorald/sonar/ProjectScanner.java

```diff
@@ -61,10 +63,11 @@ private ProjectScanner() {}
                 e.printStackTrace();
             }
         }
-
-        // TODO generalize to not directly use the SonarStaticAnalyzer
-        var violations =
-                new SonarStaticAnalyzer().findViolations(baseDir, filesToScan, rules, classpath);
+        ServiceLoader<StaticAnalyzer> analyzers = ServiceLoader.load(StaticAnalyzer.class);
+        Set<RuleViolation> violations = new HashSet<>();
+        for (StaticAnalyzer analyzer : analyzers) {
+            violations.addAll(analyzer.findViolations(baseDir, filesToScan, rules, classpath));
```

Comment:
Just one question. Do you plan on explicitly providing the name of the SPI (SonarStaticAnalyzer in this case) or keep this loop so that we report all violations from all static analyzers?

Reply:
We need another method in the future `public static Set<RuleViolation> scanProject(
            File target, File baseDir, List<Rule> rules, List<String> classpath, String analyzerID)` with `analyzerID` as parameter for supporting the explicit invocation of a single analyzer.

Reply:
That sounds great. `analyzerID` will somehow be passed through CLI and then we can parse it and pass it to service loaders.